Validate creator-principal-name annotation when creating cluster/project (#501)
rancher/webhook: pkg/resources/management.cattle.io/v3/cluster/Cluster.md

> ## Cluster

Reviewer: We don't need to specify cluster here because that gets generated in docs.md
```suggestion
## Validation Checks
```

Author: Agreed.

### Annotations validation

> When a cluster is created if `field.cattle.io/creator-principal-name` annotation is set then `field.cattle.io/creatorId` annotation must be set as well and the user's principal name in the former should match the creator's user principal id.

Reviewer: These are two separate checks so they should be different sentences for clarity
```suggestion
When a cluster is created and `field.cattle.io/creator-principal-name` annotation is set then `field.cattle.io/creatorId` annotation must be set as well. 
The value of `field.cattle.io/creator-principal-name` should match the creator's user principal id.
```

Author: Nice rewording!
